### PR TITLE
Add controller tests

### PR DESCRIPTION
**Pull request by mustabbas**
###Change Done 

- [ ] Added RSpec as a gem to the project.
- [ ] Created a Request spec file for all controllers.
- [ ] Made sure that for each action you check:
        1. If response status was correct.
        2. If a correct template was rendered.
        3. If the response body includes correct placeholder text.

### CODE REVIEW

**Review by petudeveloper — COMMENTED** [review]
Hi @mustabbas , 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

You have submitted a project, that is not following the General requirements. Make sure that you documented your work [in a professional way](https://github.com/microverseinc/curriculum-transversal-skills/blob/main/documentation/articles/professional_repo_rules.md). 
- [ ] Please add descriptive README file to your project - please use this [template](https://github.com/microverseinc/readme-template) but remember to customize it to your project.

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

File: spec/requests/posts_spec.rb [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+      get user_post_path(1, id: 1)
+      expect(response).to render_template(:show)
+    end
+
+    it 'works! reponse body includes <h1>Posts#index</h1> ' do
+      get user_post_path(1, id: 1)
+      expect(response.body).to include('<h1>Posts#index</h1>')
+    end
+  end
+end
```
Comment (DeliceLydia) [inline spec/requests/posts_spec.rb]: - [ ] Some of your tests are failing please fix them
fail

File: README.md [hunk 2]
```diff
@@ -1,24 +1,37 @@
...
+
+Feel free to check the [issues page](https://github.com/mustabbas/blog/issues).
+
+## Show your support
+
+Give a ⭐️ if you like this project!
+
+## 📝 License
+
+This project is [MIT](./MIT.md) licensed.
```
Comment (DeliceLydia) [inline README.md]: - [ ] I suggest that you make your README descriptive you can follow this [template](https://github.com/microverseinc/readme-template)

Comment (DeliceLydia) [reply README.md]: - [ ] Remember to add how to run tests locally

**Review by DeliceLydia — CHANGES_REQUESTED** [review]
## Hi @mustabbas, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there! here are highlights for your project:
- Followed git workflow
- No linter errors


## Required Changes ♻️
_Check the comments under the review._



### Cheers and Happy coding!👏👏👏

Feel free to leave a question, or a comment on the PR thread if something is not clear 100%. 

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Emmyn5600 — CHANGES_REQUESTED** [review]
Hi @mustabbas 

Great work on making the project requirements 👏 
You've done well implementing some of the project requirements, but there are still some which aren't addressed yet, please work on it and ask for another review.

 ## Required Changes ♻️

- As requested by the previous reviewers, kindly work on the readme file. Use[ the Microverse template](https://github.com/microverseinc/readme-template) to make a descriptive readme file.

## Optional suggestions

- n/a

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

#### _**Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise.**_
------------------------------------------------------------------------------------------------------------------------------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using this [form.](https://airtable.com/shrQAqnBwek5a0O0s)

**Review by arslanbisharat — APPROVED** [review]
# Approved 👏 

Hi there, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

To Highlight: 
- The Pull Request has a proper title and description. ✔️ 
- You have added a descriptive Readme file. Good Job. ✔️ 
- There is no linter error is present. ✔️ 
- Project has met all the technical aspects you needed for this project. Keep rocking! ✔️ 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of code reviews per project which you can see from your dashboard. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)_

**Pull request merged**